src/stan/callbacks/json_writer.hpp

```diff
@@ -287,12 +286,12 @@ class json_writer {
   void end_record() {
     if (output_ == nullptr)
```

Comment:
Why is this using a pointer rather than a reference?  Is the idea that you can supply a `nullptr` to turn off writing?

Reply:
> Is the idea that you can supply a nullptr to turn off writing?

Yes if it's not initialized then we don't want to allow writes